packages/expo/src/ImagePicker/ExponentImagePicker.web.ts

```diff
@@ -1,5 +1,83 @@
+import uuidv4 from 'uuid/v4';
+
+import { CameraOptions, ImageLibraryOptions, ImageResult } from './ImagePicker.types';
+import MediaTypeOptions from './MediaTypeOptions';
+
+const MediaTypeInput = {
+  [MediaTypeOptions.All]: 'video/*,image/*',
+  [MediaTypeOptions.Images]: 'image/*',
+  [MediaTypeOptions.Videos]: 'video/*',
+};
+
 export default {
   get name(): string {
     return 'ExponentImagePicker';
   },
+  async launchImageLibraryAsync({
+    mediaTypes,
+    allowsMultipleSelection = false,
+  }: ImageLibraryOptions = {}): Promise<ImageResult> {
+    return await openFileBrowser({
+      mediaTypes: MediaTypeInput[mediaTypes || MediaTypeOptions.All],
+      allowsMultipleSelection,
+    });
+  },
+  async launchCameraAsync({
+    mediaTypes,
+    allowsMultipleSelection = false,
+  }: CameraOptions = {}): Promise<ImageResult> {
+    return await openFileBrowser({
+      mediaTypes: MediaTypeInput[mediaTypes || MediaTypeOptions.All],
+      allowsMultipleSelection,
+      capture: true,
+    });
+  },
 };
+
+function openFileBrowser({
+  mediaTypes,
+  capture = false,
+  allowsMultipleSelection = false,
+}): Promise<ImageResult> {
+  const input = document.createElement('input');
+  input.style.display = 'none';
+  input.setAttribute('type', 'file');
+  input.setAttribute('accept', mediaTypes);
+  input.setAttribute('id', uuidv4());
+  if (allowsMultipleSelection) {
+    input.setAttribute('multiple', 'multiple');
+  }
+  if (capture) {
+    input.setAttribute('capture', 'camera');
+  }
+  document.body.appendChild(input);
+
+  return new Promise((resolve, reject) => {
+    input.addEventListener('change', () => {
+      if (input.files) {
+        const targetFile = input.files[0];
+        const reader = new FileReader();
+        reader.onerror = reject;
```

Comment:
Don't assign `reject` directly to error/abort and other event handlers since a DOM event object is passed in rather than an error. Need to provider handlers that convert events to Errors since we want to consistently reject (i.e. throw asynchronously) with Errors.